javascript/net/grpc/web/generator/grpc_generator.cc

```diff
@@ -904,8 +900,7 @@ void PrintProtoDtsMessage(Printer* printer, const Descriptor* desc,
     }
     vars["js_field_name"] = js_field_name;
     vars["js_field_type"] = AsObjectFieldType(field, file);
-    if ((field->type() != FieldDescriptor::TYPE_MESSAGE && !field->has_optional_keyword()) ||
```

Comment:
Trying to understand all the cases is literally giving me a headache.. 😅

so I asked ChatGPT if the before and after is equivalent and it says yet — so i trust it (and you) :D

I honestly wouldn't be surprised if the code before has bugs which would be fixed by the new code.

thanks so much for the simplification.. so much nicer :)